### PR TITLE
[Backport 2024.01.xx] Dependency updates for mapstore-services and POMs refactoring (#10015)

### PR DESCRIPTION
**Pull request by offtherailz**
[Backport 2024.01.xx] Dependency updates for mapstore-services and POMs refactoring (#10015)

Includes also backport fixes to the github actions and release procedure (so they are executed on stable branch correctly on the merge): 

- https://github.com/geosolutions-it/MapStore2/pull/10033
- https://github.com/geosolutions-it/MapStore2/pull/10036

Draft because requires this backport to be merged first. https://github.com/geosolutions-it/geostore/pull/343

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,34 +1,322 @@
...
+    <name>MapStore Root</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <mapstore2.version>DEV</mapstore2.version>
+        <tomcat.version>8.5.69</tomcat.version>
+        <tomcat.port>8080</tomcat.port>
+        <geostore-webapp.version>2.2-SNAPSHOT</geostore-webapp.version>
+        <print-lib.version>2.4-SNAPSHOT</print-lib.version>
+        <http_proxy.version>1.6-SNAPSHOT</http_proxy.version>
```
Comment (offtherailz) [inline pom.xml:17]:
```suggestion
        <geostore-webapp.version>2.1-SNAPSHOT</geostore-webapp.version>
        <print-lib.version>2.4-SNAPSHOT</print-lib.version>
        <http_proxy.version>1.5-SNAPSHOT</http_proxy.version>
```

File: project/standard/templates/web/pom.xml [hunk 2]
```diff
@@ -15,14 +15,25 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <tomcat.version>8.5.69</tomcat.version>
     <tomcat.port>8080</tomcat.port>
+    <mapstore-services.version>1.8-SNAPSHOT</mapstore-services.version>
+    <geostore-webapp.version>2.1-SNAPSHOT</geostore-webapp.version>
```
Comment (tdipisa) [inline project/standard/templates/web/pom.xml:19]: I think MS master should goes to 2.2-SNAPSHOT now for this pom file @offtherailz.

File: project/standard/templates/web/pom.xml [hunk 3]
```diff
@@ -15,14 +15,25 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <tomcat.version>8.5.69</tomcat.version>
     <tomcat.port>8080</tomcat.port>
+    <mapstore-services.version>1.8-SNAPSHOT</mapstore-services.version>
```
Comment (tdipisa) [inline project/standard/templates/web/pom.xml:18]:
```suggestion
    <mapstore-services.version>1.7-SNAPSHOT</mapstore-services.version>
```

**Review by tdipisa — COMMENTED** [review]
@offtherailz [this file](https://github.com/geosolutions-it/MapStore2/pull/10015/files#diff-3599ba7c628c7c764665046828bad74c0c8576aad03f5497cf426b59010a6d07) seems missing.

**Comment by offtherailz** [comment]
> @offtherailz [this file](https://github.com/geosolutions-it/MapStore2/pull/10015/files#diff-3599ba7c628c7c764665046828bad74c0c8576aad03f5497cf426b59010a6d07) seems missing. See also below for other things to che checked/fixed.

This changes are present on the merge branch. Probably it has  been ported in a previeous backport and it is not shown in this diff.

https://github.com/offtherailz/MapStore2/blob/bp_10015/docs/developer-guide/mapstore-migration-guide.md

File: product/pom.xml [hunk 4]
```diff
@@ -1,490 +1,440 @@
...
-            <!-- Override only if necessary -->
-            <tomcat.port>${env.MAPSTORE_BACKEND_PORT}</tomcat.port>
-        </properties>
-    </profile>
-    <profile>
-      <id>printing</id>
-      <activation>
-        <activeByDefault>false</activeByDefault>
-      </activation>
-      <build>
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply product/pom.xml:315]: The diff lines looks strange. 315 jump to 16.
Anyway I verified that that is effectively present in line 16 of the file

https://github.com/offtherailz/MapStore2/blob/bp_10015/product/pom.xml#L17C8-L23C22

Comparing with Master, the only difference looks to be the parent project version. 

master: 
https://github.com/geosolutions-it/MapStore2/blob/master/product/pom.xml

branch:
https://github.com/offtherailz/MapStore2/blob/bp_10015/product/pom.xml

[the remark this answers is not in the document]
Comment (offtherailz) [reply product/pom.xml:315]: ok, I got it. The diff calculation somehow messes up things,
but here the are the lines you needed.

https://github.com/geosolutions-it/MapStore2/pull/10105/files#diff-f9bdfc59d49013ad57196e3afa6e1eb15fe1277662806a0b66650e5f8ed05255R10

**Review by tdipisa — APPROVED** [review]

**Pull request merged**